ecs-init/exec/iptables/iptables.go

```diff
@@ -189,18 +203,51 @@ func getBlockIntrospectionOffhostAccessInputChainArgs() []string {
 	return []string{
 		"INPUT",
 		"-p", "tcp",
-		"-i", getOffhostIntrospectionInterface(),
+		"-i", defaultOffhostIntrospectionInterface,
 		"--dport", agentIntrospectionServerPort,
 		"-j", "DROP",
 	}
 }
 
-func getOffhostIntrospectionInterface() string {
+func getOffhostIntrospectionInterface() (string, error) {
 	s := os.Getenv(offhostIntrospectonAccessInterfaceEnv)
 	if s != "" {
-		return s
+		return s, nil
+	}
+	return getDefaultNetworkInterfaceIPv4()
+}
+
+// Parse /proc/net/route file and retrieves a non-loopback default network interface for IPv4 (which maps to default 0.0.0.0/0 destination)
+// Example file content:
+// $ sudo cat /proc/net/route
+// Iface   Destination Gateway     Flags   RefCnt  Use Metric  Mask   MTU Window  IRTT
+// ens5    00000000    01201FAC    0003    0   		0   512 00000000    0   0   	0
+// ...
+//
+// 1st column contains interface name
+// 2nd column contains destination network in hex
+var getDefaultNetworkInterfaceIPv4 = func() (string, error) {
+	input, err := os.Open(ipv4RouteFile)
+	if err != nil {
+		return "", fmt.Errorf("could not get IPv4 route input: %v", err)
+	}
+	defer input.Close()
+	return scanIPv4RoutesForDefaultInterface(input)
```

Comment:
Will this fail if ipv4 is disabled?

Reply:
I suppose user can explicitly delete the default IPv4 route (or maybe there's a way to tell kernel not to populate the route table on boot?). In such case it's not so meaningful setting up the drop rule anyways (as it's targeting IPv4 only). Maybe we can just log an error and move on? What do you think?

Reply:
That's true I didn't realize this was only for the offhost introspection drop rule anyways. I guess I only wonder that if it fails should we fall back to the previous behavior (harcoded eth0) in the rare case that it might affect some customer with a special routing setup that's currently working.

Reply:
Yeah okay I guess I don't see any harms doing that. 

Creating a drop rule with `-i eth0` on an instance without eth0 device actually works, although the rule probably never will be picked up.